cli/medperf/utils.py

```diff
@@ -237,23 +243,39 @@ def combine_proc_sp_text(proc: spawn) -> str:
             logging.error("Process timed out")
             raise ExecutionError("Process timed out")
         line = line.decode("utf-8", "ignore")
-        if line:
+
+        if not line:
+            continue
+
+        match = log_pattern.match(ansi_escape_pattern.sub('', line))
+        line_should_be_filtered = False
+        if match:
+            line_pid, matched_log_level_str, content = match.groups()
+            matched_log_level = logging.getLevelName(matched_log_level_str)
+
+            # if line matches conditions, it is just logged to debug; else, shown to user
+            line_should_be_filtered = (line_pid == str(proc.pid)  # hide only `mlcube` framework logs
+                                       and isinstance(matched_log_level, int)
```

Comment:
I suggest we have the logic of "deciding if we want to filter" be encapsulated in a separate function

Reply:
Agree, fixed